Warning about `struct DungeonParams` when building Minetest (mapgen/mapgen.cpp.o)

##### Issue type
<!-- Pick one below and delete others -->
 - Build issue

##### Minetest version
<!--
Paste Minetest version between quotes below
If you are on a devel version, please add git commit hash
You can use `minetest --version` to find it.
-->
Latest master/head (df6670b28a2f4e1f6094b617a25792a70400b66d as of writing this)

##### OS / Hardware
<!-- General information about your hardware and operating system -->
Operating system: Linux 4.20.3-arch1-1-ARCH x86_64
cmake version 3.13.2

##### Summary
The following warning and note appear during build.

```
[ 40%] Building CXX object src/CMakeFiles/minetest.dir/mapgen/mapgen.cpp.o
/home/dirk/mt5/minetest/src/mapgen/dungeongen.cpp: In constructor 'DungeonGen::DungeonGen(const NodeDefManager*, GenerateNotifier*, DungeonParams*)':
/home/dirk/mt5/minetest/src/mapgen/dungeongen.cpp:54:34: warning: 'void* memcpy(void*, const void*, size_t)' writing to an object of type 'struct DungeonParams' with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
   memcpy(&dp, dparams, sizeof(dp));
                                  ^
In file included from /home/dirk/mt5/minetest/src/mapgen/dungeongen.cpp:21:
/home/dirk/mt5/minetest/src/mapgen/dungeongen.h:41:8: note: 'struct DungeonParams' declared here
 struct DungeonParams {
        ^~~~~~~~~~~~~
```

##### Steps to reproduce
<!-- For bug reports or build issues, explain how the problem happened -->
Create directory to work in and du a clean build as per instructions.
```
mkdir mt5 ; cd mt5
git clone --depth 1 https://github.com/minetest/minetest.git
cd minetest
git clone --depth 1 https://github.com/minetest/minetest_game.git games/minetest_game
cmake . -DRUN_IN_PLACE=TRUE
make -j 8
```

--- a/src/mapblock.cpp
+++ b/src/mapblock.cpp
@@ -558,8 +558,9 @@ void MapBlock::deSerializeNetworkSpecific(std::istream &is)
 {
 	try {
 		const u8 version = readU8(is);
-		//if(version != 1)
-		//	throw SerializationError("unsupported MapBlock version");
+		if (version != 1) {
+			// throw SerializationError("unsupported MapBlock version");
+		}
 
 	} catch(SerializationError &e) {
 		warningstream<<"MapBlock::deSerializeNetworkSpecific(): Ignoring an error"

--- a/src/mapblock.cpp
+++ b/src/mapblock.cpp
@@ -557,10 +557,10 @@ void MapBlock::deSerialize(std::istream &is, u8 version, bool disk)
 void MapBlock::deSerializeNetworkSpecific(std::istream &is)
 {
 	try {
-		const u8 version = readU8(is);
-		if (version != 1) {
-			// throw SerializationError("unsupported MapBlock version");
-		}
+		readU8(is);
+		//const u8 version = readU8(is);
+		//if (version != 1)
+			//throw SerializationError("unsupported MapBlock version");
 
 	} catch(SerializationError &e) {
 		warningstream<<"MapBlock::deSerializeNetworkSpecific(): Ignoring an error"

--- a/src/mapgen/dungeongen.cpp
+++ b/src/mapgen/dungeongen.cpp
@@ -51,7 +51,7 @@ DungeonGen::DungeonGen(const NodeDefManager *ndef,
 #endif
 
 	if (dparams) {
-		memcpy(&dp, dparams, sizeof(dp));
+		dp = *dparams;
 	} else {
 		// Default dungeon parameters
 		dp.seed = 0;

